packages/next/server/base-server.ts

```diff
@@ -689,11 +670,14 @@ export default abstract class Server {
       )
     }
 
-    const page: { name?: string; params?: { [key: string]: string } } = {}
+    const page: RequestData['page'] = {}
     if (await this.hasPage(params.parsedUrl.pathname)) {
       page.name = params.parsedUrl.pathname
     } else if (this.dynamicRoutes) {
+      const isApi = isApiRoute(params.parsedUrl.pathname)
       for (const dynamicRoute of this.dynamicRoutes) {
+        if (isApi && !isApiRoute(dynamicRoute.page)) continue
```

Comment:
I think it might be worth to leave a comment of why this is like that. If I'm not wrong it is because we want to treat in a special way API Routes like we already do with pages.

Reply:
done